### PR TITLE
machines: fix detection of the nic source when displaying the warning in the modal

### PR DESCRIPTION
**Pull request by KKoukiou**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1911657

### CODE REVIEW

**Comment by KKoukiou** [comment]
Whoever reviews this I need an advice for how to use NICAddDialog in order that I avoid all this test_obj passing in order to access the self.browser from MachinesCase and the self.assertEqual from unittest

**Comment by KKoukiou** [comment]
Gr why I can't use these methods from netlib :'( ?

File: test/verify/check-machines-nics [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import parent
 from machineslib import *
 from testlib import *
+from netlib import  *
```
Comment (KKoukiou) [inline test/verify/check-machines-nics:23]: Oh this is redundant.

**Review by martinpitt — APPROVED** [review]
Thanks! The `self.NICAddDialog(self)` syntax is rather weird, and I think it ought to be possible to drop the first `self.`. But this is not important at all, the bug fix is.

**Pull request merged**